File: Components/Reporting/VB/ASPxReportDemos/App_Code/DemoLoggingService.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.Utils.About
Imports DevExpress.XtraReports.Web.Native.ClientControls.Services

Namespace DevExpress.Web.Demos
	Public Class DemoLoggingService
		Inherits ExtendedLoggingService
		Public Sub New()
			MyBase.New(Function() DefaultLoggingService.Instance)
		End Sub
		Public Overrides Sub LogException(ByVal exception As Exception, ByVal message As String)
			MyBase.LogException(exception, message)
			UAlgo.Default.DoEventException(exception)
		End Sub
	End Class
End Namespace

File: Components/Reporting/CS/MVCxReportDemos/Code/DemoLoggingService.cs
using System;
using DevExpress.Utils.About;
using DevExpress.XtraReports.Web.Native.ClientControls.Services;

namespace DevExpress.Web.Demos {
    public class DemoLoggingService : ExtendedLoggingService {
        public DemoLoggingService() : base(() => DefaultLoggingService.Instance) { }
        public override void LogException(Exception exception, string message) {
            base.LogException(exception, message);
            UAlgo.Default.DoEventException(exception);
        }
    }
}
